lib/util/LocationStatusManager.js

```diff
@@ -238,9 +238,10 @@ class LocationStatusManager {
      * in mongo based on local values
      * @param {string} location location name
      * @param {callback} cb callack
-     * @returns {undefined}
+     * @returns {undefined}z
      */
     _updateServiceStatusForLocation(location, cb) {
+        const previousState = JSON.parse(JSON.stringify(this._locationStatusStore[location].getValue()));
```

Comment:
`this._locationStatusStore[location].getValue()` Already returns a JSON no need to stringify and then parse ?

Reply:
Actually, this is not really the previous state, if you look at the Mongo query you'll see that it's the value we use to update the document. The previous state can be acquire in the caller of this function.